### PR TITLE
Include lightning-persister in generated bindings

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Built on #11.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Rebased and bumped to latest upstream.

File: genbindings.sh [hunk 1]
```diff
@@ -43,17 +43,38 @@ OUT_F="$(pwd)/lightning-c-bindings/include/rust_types.h"
...
-	RUST_BACKTRACE=1 "$BIN" "$OUT/" lightning "$OUT_TEMPL" "$OUT_F" "$OUT_CPP"
+pushd "$LIGHTNING_PATH/lightning"
+RUSTC_BOOTSTRAP=1 cargo rustc $FEATURES_ARGS --profile=check -- -Zunstable-options --pretty=expanded > /tmp/lightning-crate-source.txt
 popd
 
+if [ "$HOST_PLATFORM" = "host: x86_64-apple-darwin" ]; then
+	# OSX sed is for some reason not compatible with GNU sed
+	sed -E -i '' 's/#!\[crate_name = "(.*)"\]/pub mod \1 {/' /tmp/lightning-crate-source.txt
+else
+	sed -E -i 's/#!\[crate_name = "(.*)"\]/pub mod \1 {/' /tmp/lightning-crate-source.txt
```
Comment (arik-so) [inline genbindings.sh:54]: hitting this path with the following error:

+ sed -E -i 's/#!\[crate_name = "(.*)"\]/pub mod \1 {/' /tmp/lightning-crate-source.txt
sed: 1: "/tmp/lightning-crate-so ...": extra characters at the end of l command

File: genbindings.sh [hunk 2]
```diff
@@ -43,17 +43,38 @@ OUT_F="$(pwd)/lightning-c-bindings/include/rust_types.h"
...
+	popd
+	sed -i '1ipub mod lightning_persister {\n' /tmp/lightning-persist-crate-source.txt
+	echo "}" >> /tmp/lightning-persist-crate-source.txt
+	cat /tmp/lightning-persist-crate-source.txt >> /tmp/lightning-crate-source.txt
+	rm /tmp/lightning-persist-crate-source.txt
+fi
+
+cat /tmp/lightning-crate-source.txt | RUST_BACKTRACE=1 "$BIN" "$OUT/" "$OUT_TEMPL" "$OUT_F" "$OUT_CPP"
+
 HOST_PLATFORM="$(rustc --version --verbose | grep "host:")"
```
Comment (arik-so) [inline genbindings.sh:70]: move this up before line 50

**Review by arik-so — CHANGES_REQUESTED** [review]

File: genbindings.sh (hunk 1 above)
Comment (TheBlueMatt) [reply genbindings.sh:54]: Hmmm, I don't know that I can really debug this as I don't have a Mac - is there any way you could try to figure out how to call sed with a regex and references on OSX?

Comment (TheBlueMatt) [reply genbindings.sh:55]: Oops, I misunderstood your second comment, can you take a look at the freshly-pushed version?

**Comment by TheBlueMatt** [comment]
Ok, should be fixed now, though its now based on #13.

**Review by arik-so — APPROVED** [review]

**Pull request merged**